src/contexts/bridge.context.tsx

```diff
@@ -373,6 +402,59 @@ const BridgeProvider: FC = (props) => {
     [getTokenPrice, getToken]
   );
 
+  const REFRESH_PAGE_SIZE = 100;
+
+  const refreshBridges = useCallback(
+    async ({ env, ethereumAddress, quantity }: RefreshBridgesParams): Promise<Bridge[]> => {
+      refreshCancelTokenSource.current = axios.CancelToken.source();
+      const completePages = Math.floor(quantity / REFRESH_PAGE_SIZE);
+      const remainderBridges = quantity % REFRESH_PAGE_SIZE;
+      const requiredRequests = remainderBridges === 0 ? completePages : completePages + 1;
+      return (
+        await Promise.all(
+          Array(requiredRequests)
+            .fill(null)
+            .map((_, index) => {
+              const offset = index * REFRESH_PAGE_SIZE;
+              const isLast = index + 1 === requiredRequests;
+              const isRemainderRequestRequired = isLast && remainderBridges !== 0;
+              const limit = isRemainderRequestRequired ? remainderBridges : REFRESH_PAGE_SIZE;
+              return getBridges({
+                env,
+                ethereumAddress,
+                limit,
+                offset,
+                cancelToken: refreshCancelTokenSource.current.token,
+              });
+            })
+        )
+      ).reduce((acc, curr) => [...acc, ...curr], []);
+    },
+    [getBridges]
+  );
+
+  const fetchBridges = useCallback(
```

Comment:
What is the point of this `fetchBridges` function instead of calling `getBridges` and `refreshBridges` from the outside?

Reply:
Taking care of cancelling the previous requests:

`refreshCancelTokenSource.current.cancel()`

Reply:
But we can do the same inside the `refreshBridges` function right?

Reply:
mmm... no, we need to do it before calling `getBridges` when `params.type === "load"`.